ipmikvm

```diff
@@ -319,10 +347,12 @@ get_app_jar() {
     umask 0077  # storing latest.jnlp here; might contain temp passwords
     test -d "$app_cache_dir" || mkdir -p "$app_cache_dir" || exit 1
     echo "$launch_jnlp" >"$app_cache_dir/latest.jnlp"
+    flock $LOCKFD
```

Comment:
Why not `exec $LOCKFD<>"$app_cache_dir/lock"` here?

Reply:
You mean to be more "lazy"?

Yeah, I could do that 👍

Reply:
Not only lazy. With `$app_cache_dir` instead of `$APP_CACHE_DIR` the lock is for the _specific_ app cache, not the total (root) app cache.

If we look here:
```sh
JAR=$(get_app_jar "$JNLP" "$APP_CACHE_DIR")
```
and:
```sh
get_app_jar() {
    local launch_jnlp="$1"
    local app_cache_dir="$2"
# ...
    *)
        app_cache_dir="$app_cache_dir/$jar_base"
# ...
```
We see that the "$app_cache_dir" is specific to the jar_base, which could be `iKVM__V1.69.31.0x0`. For that jar_base, we might want locking. It's fine for others to download simultaneously.

Also, minor nit:
`exec 4<>$APP_CACHE_DIR/lock` is lacking the double quotes:
```console
$ APP='a b v'

$ exec 4<>$APP
bash: $APP: ambiguous redirect

$ exec 4<>"$APP"
```